### ISSUE TITLE
Short ISIs - peak at center of autocorrelogram

### ISSUE THREAD

**Issue by niranjankambi** [MouseLand/Kilosort2#158]
Hi, 
I have this nicely sorted unit (see the pic attached below) in channel 8. The autocorrelogram, however, has a peak at center which, IMO, means that the spikes included in this unit have short ISIs. Any clue as to how to get rid of them is much appreciated.
Thanks,
Niranjan
![Screenshot from 2019-12-13 17-44-36](https://user-images.githubusercontent.com/14698586/70839245-664e4f80-1dd1-11ea-9668-1ec7f7244df0.png)

**Comment by angelonc** [comment]
This does not quite look nicely sorted to me, in the FeatureView and Waveform view you can see that you have many large amplitude artifacts contaminating this cluster. You can right click around your cluster and press K to discard these events, which are probably causing the autocorrelation peak sub 1ms.

**Comment by marius10p** [comment]
Sorry, this just looks like noise. Kilosort is very good at finding repeating "events", even when they are relatively small and likely noise.

**Comment by niranjankambi** [comment]
Thanks @angelonc and @marius10p for your replies. @angelonc as per your suggestion I will try and remove these artifacts manually.
RE: @marius10p suggestion, I should have a picked a better example as this channel does seem to have lot of really large artifacts.  So I was wondering if you have any specific suggestions on removing those really large artifacts before feeding the data to kilosort.

**Comment by marius10p** [comment]
The artifacts may be large, but they probably only account for a minority of events detected by KIlosort2. Usually people write their own artifact removal scripts to pre-process the data before putting into KIlosort2, because many artifacts looks very different, so you have to treat them differently.

**Comment by niranjankambi** [comment]
Thanks @marius10p as I am trying a few routines to see if I can get rid of the large sudden artifacts. However, going to back to my original question - is there a way, if any, to remove short ISI spikes in phyGUI? If not, is there a parameter in the kilosort GUI or associated scripts where we can specify it? Thanks,
N

**Comment by niranjankambi** [comment]
See this channel, for example, where we do see a clear and large waveform in channel 4. However, the autocorrelogram still has a significant peak at the center indicating possibly many short ISIs
![image](https://user-images.githubusercontent.com/14698586/71489863-6b59c880-27ed-11ea-8494-e3c19cc6a602.png)

**Comment by marius10p** [comment]
It's not surprising, that is a relatively small waveform that is corrupted by background noise, hence the ISI violations. See if you can clean up the unit, but that just might be the best you can do. It would be wrong to selectively remove the relatively few short ISIs, because those are only a symptom of this problem. Many, many more contaminating spikes will not be coincident with your unit, and you wouldn't be able to remove them this way.

**Comment by niranjankambi** [comment]
@marius10p thanks for your reply and suggestion. I will try to clean up the unit and see if that makes a difference.
One more thing that I noticed is that short ISIs are observed in spikes which have an inverted waveform more than in the ones which are detected as downward dipping. Not sure if you have seen anything like this.
 I also noticed that our criteria for classifying size of the spike waveform vary. May be it is due to the type of probes I use for recording that makes the difference. I record using laminar probes from Microprobes and not sure if anyone else has tried sorting data with ks2 from these probes.
